lib/puppet/pops/validation.rb

```diff
@@ -445,22 +445,14 @@ def prune(&block)
 
     private
 
-    def ignore diagnostic
+    def _accept(diagnostic)
       @diagnostics << diagnostic
-    end
-
-    def error diagnostic
-      @diagnostics << diagnostic
-      @error_count += 1
-    end
-
-    def warning diagnostic
-      @diagnostics << diagnostic
-      @warning_count += 1
-    end
-
-    def deprecation diagnostic
-      warning diagnostic
+      case diagnostic.severity
+      when :error
+        @error_count += 1
+      when :deprecation, :warning
+        @warning_count += 1
```

Comment:
The `:deprecation` case used to call `warning` and didn't increment the `warning_count

Reply:
Right. And the call to `warning` in `deprecation` would increment the `@warning_count`, right?

Reply:
Oh that's confusing. I thought the previous call to `warning deprecation` was calling the puppet log function named `warning`. But what you have makes sense.